core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/telemetry/TelemetryAkkaService.scala

```diff
@@ -0,0 +1,123 @@
+
+import akka.actor.ActorSystem
+
+import io.circe.Json
+
+import scala.concurrent.ExecutionContextExecutor
+import scala.concurrent.duration.Duration
+
+import org.slf4j.{Logger, LoggerFactory}
+
+import java.util.concurrent.TimeUnit
+
+import com.snowplowanalytics.iglu.core.SelfDescribingData
+import com.snowplowanalytics.snowplow.scalatracker.Emitter.EndpointParams
+import com.snowplowanalytics.snowplow.scalatracker.Tracker
+import com.snowplowanalytics.snowplow.scalatracker.idimplicits._
+import com.snowplowanalytics.snowplow.scalatracker.emitters.id.SyncEmitter
+import com.snowplowanalytics.snowplow.scalatracker.Emitter._
+import com.snowplowanalytics.snowplow.collectors.scalastream.model.{
+  CollectorConfig,
+  GooglePubSub,
+  Kinesis,
+  Sqs,
+  TelemetryConfig
+}
+
+/** Akka implementation of telemetry service.
+  *
+  * @param teleCfg - telemetry configuration
+  * @param cloud - cloud vendor
+  * @param region - deployment region
+  * @param appName - application name as defined during build (take it from BuildInfo)
+  * @param appVersion - application name as defined during build (take it from BuildInfo)
+  */
+case class TelemetryAkkaService(
+  teleCfg: TelemetryConfig,
+  cloud: Option[CloudVendor],
+  region: Option[String],
+  appName: String,
+  appVersion: String
+) {
+  private lazy val log: Logger = LoggerFactory.getLogger(getClass)
+
+  private lazy val payload: SelfDescribingData[Json] = makeHeartbeatEvent(teleCfg, cloud, region, appName, appVersion)
+
+  def start()(implicit actorSystem: ActorSystem): Unit =
+    if (teleCfg.isDisabled) {
+      log.info(s"Telemetry disabled")
+    } else {
+      log.info(s"Telemetry enabled")
+      val scheduler                                   = actorSystem.scheduler
+      implicit val executor: ExecutionContextExecutor = actorSystem.dispatcher
+
+      def emitterCallback(params: EndpointParams, req: Request, res: Result): Unit =
+        res match {
+          case Result.Success(_) => log.debug(s"telemetry send successfully")
+          case Result.Failure(code) =>
+            log.warn(s"Scala telemetry tracker got unexpected HTTP code $code from ${params.getUri}")
+          case Result.TrackerFailure(exception) =>
+            log.warn(
+              s"Scala telemetry tracker failed to reach ${params.getUri} with following exception $exception" +
+                s" after ${req.attempt} attempt"
+            )
+          case Result.RetriesExceeded(failure) =>
+            log.warn(s"Scala telemetry tracker has stopped trying to deliver payload after following failure: $failure")
+        }
+
+      val emitter: SyncEmitter = SyncEmitter(
+        EndpointParams(teleCfg.url, port = teleCfg.port, https = teleCfg.isSecure),
+        callback = Some(emitterCallback)
+      )
+      // telemetry - Unique identifier for website / application (aid)
+      // root - The tracker namespace (tna)
+      val tracker = new Tracker(NonEmptyList.of(emitter), "root", "telemetry")
```

Comment:
```suggestion
      val tracker = new Tracker(NonEmptyList.of(emitter), "tracker-telemetry", appName)
```

See [here](https://github.com/snowplow/enrich/blob/feature/file-sink-rotate/modules/common-fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/common/fs2/Telemetry.scala#L79)

Reply:
Reference code sets application id to `root` and tracker namespace to `appName`.  Please elaborate.

Reply:
[This is](https://github.com/snowplow/snowplow-scala-tracker/blob/master/modules/core/src/main/scala/com.snowplowanalytics.snowplow/scalatracker/Tracker.scala#L72-L73) how these 2 fields are used in the tracker.

And this is [what these fields mean](https://docs.snowplowanalytics.com/docs/collecting-data/collecting-from-own-applications/snowplow-tracker-protocol/#common-parameters-platform-and-event-independent).

I think that all telemetry trackers across apps should use the same namespace and then each app should identify itself.

Reply:
There is already provided the user-defined name for the application, which is a part of the Context.
https://github.com/snowplow/iglu-central/blob/f163f9565c0a91e3dc1c6ae3e56d4ced7f30bae2/schemas/com.snowplowanalytics.oss/oss_context/jsonschema/1-0-1#L76

So suggestion is to use the following conversion:
- aid - telemetry (constant)
- tna - name of component family - collector, enrich

Reply:
> I think that all telemetry trackers across apps should use the same namespace and then each app should identify itself.

That sounds sensible.

> There is already provided the user-defined name for the application, which is a part of the Context.

Yes true, but with a difference in meaning.  The value in the context describes the _application we are describing_ whereas the `aid` describes the _application sending the event_.  The distinction is important when terraform sends the event.

Currently the [terraform module sets the aid to "terraform-oss-modules"](https://github.com/snowplow-devops/terraform-snowplow-telemetry/blob/v0.2.0/main.tf#L9) and it sets tna to empty string.  The latter could be improved.

For the collector and for enrich it makes sense for `aid` to be exactly equal to the context's `applicationName`.  Because the app sending the event **is** the app being described.

For tna I slightly prefer `telemetry` rather than `tracker-telemetry`.  Not strongly opinionated.  And fairly sure the people using this data won't mind much either.

Reply:
Let's use `"telemetry"`. `tracker` seems redundant